src/main/java/demon/Demon.java

```diff
@@ -0,0 +1,128 @@
+package demon;
+
+import java.nio.file.NoSuchFileException;
+
+import java.io.IOException;
+
+import java.util.Scanner;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * A chatbot that allows user to add, delete, and modify tasks or reminders.
+ */
+public class Demon {
+    private TaskList tasks;
+    private Ui ui;
+    String filePath = "src/main/taskList.txt";
+
+    /**
+     * Initializes the bot and load tasks saved in taskList.txt file
+     * if there is any previous tasks.
+     *
+     * @param filePath The relative path to the taskList.txt file
+     */
+    public Demon(String filePath) {
+        ui = new Ui();
+        final Storage STORAGE = new Storage(filePath);
+        try {
+            tasks = new TaskList(STORAGE.load());
+        } catch (NoSuchFileException e) {
+            tasks = new TaskList();
+            ui.showLoadingError();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        ui = new Ui();
+        Commands command = new Commands(filePath);
```

Comment:
@Colex2000 with reference to the above comment, here you are using the instance variable `this.filePath` (declared in line 17).